tests/integration/src/test/scala/mesosphere/marathon/integration/ExternalVolumesIntegrationTest.scala

```diff
@@ -0,0 +1,81 @@
+package mesosphere.marathon
+package integration
+
+import mesosphere.AkkaIntegrationTest
+import mesosphere.marathon.integration.setup._
+import play.api.libs.json.Json
+
+class ExternalVolumesIntegrationTest extends AkkaIntegrationTest with EmbeddedMarathonTest {
+
+  override val marathonArgs: Map[String, String] = Map(
+    "enable_features" -> "external_volumes"
+  )
+
+  "ExternalVolume names with options" should {
+    "be considered valid" in {
+      Given("a new app")
+      val app = Json.parse(appWithExtVol).as[raml.App]
+
+      When("The app is deployed")
+      val result = marathon.createAppV2(app)
+
+      Then("The app is created")
+      result should be(Created)
+    }
+  }
+
+  val appWithExtVol: String =
```

Comment:
What is the reason not to use the RAML models here?

Reply:
I wanted to be as close as possible to the config the user tried to POST. Also, if I was to test this with a handcrafted RAML model, there'd be no guarantee that this JSON would translate into exactly that RAML model – so, again, this would not test a roundtrip but just a portion of the API, and the issue we're fixing here in this PR came up because we didn't do roundtrip testing in the first place.

IMO, an integration test should always test end to end – I can safely state that the JSON used in here is a valid example.

Reply:
Fair point.

Reply:
AppsResourceTest tests the endpoint such that serialization is also tested, without needing to spawn up Marathon, Mesos, and Zookeeper

Reply:
✅